BaseStation/SensorDiagnostics/SensorDiagnostics/LogSensors.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Scarlet;
+using Scarlet.Components;
+using Scarlet.Components.Sensors;
+using Scarlet.Utilities;
+using System.IO;
+
+namespace FlatRobot
+{
+    static class LogSensors
+    {
+        private static List<List<string>>[] fileOutput = new List<List<string>>[7];
+        private static StreamWriter SensorLogFile;
+
+        private const string delimiter = ",";
+        private const string LogFilesLocation = "Logs";
+
+        public static void logSensors()
+        {
+            int[] sensorCount = new int[7];
+            /*
+             * 0 - Encoders
+             * 1 - LimitSwtichs
+             * 2 - MAX21855s
+             * 3 - MPU6050s
+             * 4 - Potentiometers
+             * 5 - VEML6070s
+             */
+            foreach (ISensor s in Global.sensors)
+            {
+                if (s.GetType() == typeof(Scarlet.Components.Sensors.Encoder))
+                {
+                    Scarlet.Components.Sensors.Encoder temp = (Scarlet.Components.Sensors.Encoder)s;
+                    fileOutput[0][sensorCount[0]].Add(temp.Angle.ToString());
+                    sensorCount[0]++;
+                } else
+                if (s.GetType() == typeof(LimitSwitch))
+                {
+                    LimitSwitch temp = (LimitSwitch)s;
+                    temp.SwitchToggle += (object sender, LimitSwitchToggle e) => { fileOutput[1][sensorCount[2]].Add(e.CurrentState.ToString()); };
+                    sensorCount[1]++;
+
+                } else
+                if (s.GetType() == typeof(MAX31855))
+                {
+                    MAX31855 temp = (MAX31855)s;
+                    fileOutput[2][sensorCount[2]].Add(temp.GetRawData().ToString());
+                    sensorCount[2]++;
+                } else
+                if (s.GetType() == typeof(MPU6050))
```

Comment:
This kind of if structure is a good sign that polymorphism would be useful here. It might be a good idea to add a method to the objects, or use extension methods here

Reply:
I'm not overly familiar with polymorphism in c# or extension methods in general. I will look into them but any help or ideas that you have around that would be useful.